js/blocks/controls/user-input.js

```diff
@@ -0,0 +1,298 @@
+/**
+ * The UserInput component, forked from the URLInput component in Gutenberg.
+ *
+ * It would be ideal to extend that component instead of forking it.
+ * But there are changes throughout this class.
+ * For example, URLInput stores the URL and post, and this only stores the user slug.
+ * This still depends on styling from URLInput, including the class editor-url-input__suggestion.
+ *
+ * @see https://github.com/WordPress/gutenberg/blob/0ede174e6ff482085ee51b6a99bea0801c11d609/packages/editor/src/components/url-input/index.js
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import scrollIntoView from 'dom-scroll-into-view';
+
+/**
+ * WordPress dependencies
+ */
+const { __, sprintf, _n } = wp.i18n;
+const { Component, createRef } = wp.element;
+const { decodeEntities } = wp.htmlEntities;
+const { UP, DOWN, ENTER, TAB } = wp.keycodes;
+const { Spinner, withSpokenMessages, Popover } = wp.components;
+const { withInstanceId } = wp.compose;
+const apiFetch = wp.apiFetch;
+const { addQueryArgs } = wp.url;
+
+// Since UserInput is rendered in the context of other inputs, but should be
+// considered a separate modal node, prevent keyboard events from propagating
+// as being considered from the input.
+const stopEventPropagation = ( event ) => event.stopPropagation();
+
+class UserInput extends Component {
+	constructor( { autocompleteRef } ) {
+		super( ...arguments );
+
+		this.onChange = this.onChange.bind( this );
+		this.onKeyDown = this.onKeyDown.bind( this );
+		this.autocompleteRef = autocompleteRef || createRef();
+		this.inputRef = createRef();
+		this.updateSuggestions = this.updateSuggestions.bind( this );
+
+		this.suggestionNodes = [];
+
+		this.state = {
+			users: [],
+			showSuggestions: false,
+			selectedSuggestion: null,
+		};
+	}
+
+	componentDidUpdate() {
+		const { showSuggestions, selectedSuggestion } = this.state;
+		// only have to worry about scrolling selected suggestion into view
+		// when already expanded
+		if ( showSuggestions && selectedSuggestion !== null && ! this.scrollingIntoView ) {
```

Comment:
This seems pretty edge case. What would be the effect of removing this code? If we just replaced it with a `focus` event on the input, would that basically achieve the same thing?

I'm a little allergic to external dependencies, sorry.

Reply:
Good question, I'm not sure what it does. It's from `URLInput` in Gutenberg. 

At least so far, it looks like deleting `componentDidUpdate()` entirely is fine. Here's what it looks like when all of the logic in `componentDidUpdate()` is deleted:

![comp-did-update](https://user-images.githubusercontent.com/4063887/53790136-dc8d5f80-3eeb-11e9-98e8-30e1064ff864.gif)

https://github.com/getblocklab/block-lab/pull/233/commits/c464479dd004e23cbb19eda7b9c0504da4b9936d removes `componentDidUpdate()`, and https://github.com/getblocklab/block-lab/pull/233/commits/c6954ecf4b482f151e95a26faed945769f014f32 removes the dependency from `package.json`. If we later find a bug, we can address it then.